misc/bazel/cargo-gazelle/src/targets.rs

```diff
@@ -183,6 +183,144 @@ impl ToBazelDefinition for RustLibrary {
     }
 }
 
+/// [`rust_binary`](https://bazelbuild.github.io/rules_rust/defs.html#rust_binary)
+#[derive(Debug)]
+pub struct RustBinary {
+    name: Field<QuotedString>,
+    crate_root: Field<QuotedString>,
+    features: Field<List<QuotedString>>,
+    aliases: Field<Aliases>,
+    deps: Field<List<QuotedString>>,
+    proc_macro_deps: Field<List<QuotedString>>,
+    data: Field<List<QuotedString>>,
+    compile_data: Field<List<QuotedString>>,
+    env: Field<Dict<QuotedString, QuotedString>>,
+    rustc_flags: Field<List<QuotedString>>,
+    rustc_env: Field<Dict<QuotedString, QuotedString>>,
+}
+
+impl RustTarget for RustBinary {
+    fn rules(&self) -> Vec<Rule> {
+        vec![Rule::RustBinary]
+    }
+}
+
+impl RustBinary {
+    pub fn generate(
+        config: &GlobalConfig,
+        metadata: &PackageMetadata,
+        crate_config: &CrateConfig,
+        target: &BuildTarget,
+    ) -> Result<Self, anyhow::Error> {
+        let crate_root_path = metadata
+            .manifest_path()
+            .parent()
+            .ok_or_else(|| anyhow::anyhow!("crate is at the root of the filesystem?"))?;
+        let name = match target.id() {
+            BuildTargetId::Binary(name) => name,
+            x => panic!(
+                "can only generate `rust_binary` rules for binary build targets, found {x:?}"
+            ),
+        };
+        let target_name = QuotedString::new(name.to_case(Case::Snake));
+
+        let binary_path = target.path();
+        let binary_path = binary_path
+            .strip_prefix(crate_root_path)
+            .map_err(|_| anyhow::anyhow!("binary is not inside workspace?"))?;
+        let binary_path = QuotedString::new(binary_path.to_string());
+
+        // Collect all dependencies.
+        let all_deps = WorkspaceDependencies::new(config, metadata);
```

Comment:
No need to exclude dev-deps?

Reply:
We are excluding dev-deps with the `.iter(DependencyKind::Normal, ...)` below, which seems like the right thing to do because binaries can't rely on dev deps?